cylc/flow/task_events_mgr.py

```diff
@@ -587,27 +588,36 @@ def process_message(
         # Any message represents activity.
         self.reset_inactivity_timer_func()
 
-        if not self._process_message_check(
-                itask, severity, message, event_time, flag, submit_num):
+        proc_message_check = self._process_message_check(
+            itask, severity, message, event_time, flag, submit_num
+        )
+        quiet = False
+        if not proc_message_check:
             return None
+        elif proc_message_check == 'quiet':
+            quiet = True
 
         # always update the workflow state summary for latest message
         if flag == self.FLAG_POLLED:
             new_msg = f'{message} {self.FLAG_POLLED}'
         else:
             new_msg = message
-        self.data_store_mgr.delta_job_msg(
-            itask.tokens.duplicate(job=str(submit_num)).relative_id,
-            new_msg
-        )
 
         # Satisfy my output, if possible, and spawn children.
         # (first remove signal: failed/EXIT -> failed)
-
         msg0 = message.split('/')[0]
         completed_trigger = itask.state.outputs.set_msg_trg_completion(
             message=msg0, is_completed=True)
-        self.data_store_mgr.delta_task_output(itask, msg0)
+
+        if not quiet:
+            # Add information about task to data-store - if we
+            # haven't already:
+            self.data_store_mgr.delta_job_msg(
+                itask.tokens.duplicate(job=str(submit_num)).relative_id,
+                new_msg
+            )
+
+            self.data_store_mgr.delta_task_output(itask, msg0)
 
         # Check the `started` event has not been missed e.g. due to
         # polling delay
```

Comment:
I don't think this should be run for duplicate messages either.

Reply:
The whole of the rest of the method or just this check?